libminifi/src/EventDrivenSchedulingAgent.cpp

```diff
@@ -44,16 +44,8 @@ uint64_t EventDrivenSchedulingAgent::run(const std::shared_ptr<core::Processor>
       // No work to do or need to apply back pressure
       return this->bored_yield_duration_;
     }
-
-    // Block until work is available
-
-    processor->waitForWork(1000);
-
-    if (!processor->isWorkAvailable()) {
-      return 1000;
-    }
   }
-  return 0;
+  return 10;  // Let's check back for work in 10ms or when a thread is available to execute
```

Comment:
This limits the event driven processors to 100 triggers/sec, and in case they don't/can't use batching to 100 FF/s, even when we know they would have work to do. This would broke many flows out in the wild. We should find a way to solve the starvation issue without introducing a bottleneck.

Reply:
Done, processors gets at most 500ms of continuous slice before waiting for 10ms.

Reply:
Could that number be configurable?

Reply:
Yep, will add to iter 2 when unifying thread pool.

Reply:
Done